### PR TITLE
Implemented and deployed a mapped membank for the ppu

### PR DESCRIPTION
**Pull request by ml85**
This PR introduces a MappedMemBank class that can be used for memory that is mapped in the CPUs address space but resides on another chip. One such case is the PPU registers.
Unfortunately I had to disable some checks in our clang-tidy configuration due to a warning in google mock. This is a know false positive, see for example https://github.com/abseil/googletest/issues/853. Maybe you have some idea on how we could handle this in a better way?

### CODE REVIEW

File: core/src/mapped_membank.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 Robin Linden <dev@robinlinden.eu>
```
Comment (robinlinden) [inline core/src/mapped_membank.h]: We should use `Copyright 2018 Evil Corp Ltd. contributors` or something like it.

File: core/include/core/ippu.h [hunk 2]
```diff
@@ -10,6 +10,10 @@ class IPpu {
 public:
     virtual ~IPpu() = default;
 
+    virtual uint8_t read_byte(uint16_t addr) const = 0;
+
```
Comment (robinlinden) [inline core/include/core/ippu.h]: I think it's nicer when functions are grouped in some way, so maybe remove this blank line? I don't know what you guys think, but I think having one blank line per line of code is really hard to parse.

File: core/test/src/mock_ppu.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#include "core/ippu.h"
+
+#include <gmock/gmock.h>
+
+namespace n_e_s::core::test {
+
+class MockPpu : public IPpu {
+public:
+    MOCK_CONST_METHOD1(read_byte, uint8_t(uint16_t addr));
+
```
Comment (robinlinden) [inline core/test/src/mock_ppu.h]: Same grouping thing.

File: core/include/core/ippu.h (hunk 2 above)
Comment (ml85) [reply core/include/core/ippu.h]: I can, without any problem, read code that is grouped OR have one blank line per line of code. That is how good I'm. But to make life easier for you Robin I'll change this and start think about my blank lines.

**Comment by ml85** [comment]
I don't get the warning from clang-tidy when getting the tip from the master branch in googletest. How do you feel about not using a released version?

**Comment by robinlinden** [comment]
I'm fine with that. :)
